### PR TITLE
Part 2 of reviewing all list constructors and adding a size where possible.

### PR DESCRIPTION
**Pull request by KoenDG**
This PR is related to issue #24226

This is part 2, follow-up to part 1, which can be found here: #24439

I'd like to point out that `Iterable<T>` doesn't have a `size()` method.

And these are the remaining PRs I still have open: #24533 and #24340

===========

1. Contributor agreement: Has been signed

2. Contributor guidelines: I have read them.

3. Gradle check: build successful.

4. PR is against master

5. The PR is not OS specific.

6. I'm not part of a class.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/plugins/PluginsService.java [hunk 1]
```diff
@@ -91,8 +91,8 @@
     public PluginsService(Settings settings, Path modulesDirectory, Path pluginsDirectory, Collection<Class<? extends Plugin>> classpathPlugins) {
         super(settings);
 
-        List<Tuple<PluginInfo, Plugin>> pluginsLoaded = new ArrayList<>();
-        List<PluginInfo> pluginsList = new ArrayList<>();
+        List<Tuple<PluginInfo, Plugin>> pluginsLoaded = new ArrayList<>(classpathPlugins.size());
+        List<PluginInfo> pluginsList = new ArrayList<>(classpathPlugins.size());
```
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/plugins/PluginsService.java]: These aren't right, we load more than just classpath plugins, it's not worth jumping through hoops trying to size them correctly, this code is executed exactly once in the life of an Elasticsearch server or transport client instance. Let's revert this change.

File: core/src/main/java/org/elasticsearch/index/fielddata/plain/ParentChildIndexFieldData.java [hunk 2]
```diff
@@ -300,7 +300,7 @@ public long ramBytesUsed() {
 
         @Override
         public void close() {
-            List<Releasable> closeables = new ArrayList<>();
+            List<Releasable> closeables = new ArrayList<>(atomicFD.size());
```
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/index/fielddata/plain/ParentChildIndexFieldData.java]: This one is not right, it's flattening a bunch of collections, please revert.

File: core/src/test/java/org/elasticsearch/common/hppc/HppcMapsTests.java [hunk 3]
```diff
@@ -23,6 +23,7 @@
 import org.elasticsearch.test.ESTestCase;
 
 import java.util.ArrayList;
+import java.util.Collection;
```
Comment (jasontedor) [inline core/src/test/java/org/elasticsearch/common/hppc/HppcMapsTests.java]: This is not needed, please revert.

File: test/framework/src/main/java/org/elasticsearch/test/InternalAggregationTestCase.java [hunk 4]
```diff
@@ -54,9 +54,9 @@ protected T createUnmappedInstance(String name,
 
     public void testReduceRandom() {
         String name = randomAlphaOfLength(5);
-        List<T> inputs = new ArrayList<>();
-        List<InternalAggregation> toReduce = new ArrayList<>();
         int toReduceSize = between(1, 200);
+        List<T> inputs = new ArrayList<>(toReduceSize);
+        List<InternalAggregation> toReduce = new ArrayList<>(toReduceSize);
```
Comment (jasontedor) [inline test/framework/src/main/java/org/elasticsearch/test/InternalAggregationTestCase.java]: This one is not right, sometimes a random reduce is added, let's add one more here to `toReduceSize`.

**Review by jasontedor — CHANGES_REQUESTED** [review]
I left some changes.

File: test/framework/src/main/java/org/elasticsearch/test/InternalAggregationTestCase.java (hunk 4 above)
Comment (jasontedor) [inline test/framework/src/main/java/org/elasticsearch/test/InternalAggregationTestCase.java]: And to be clear, I mean shifting the random decision up here and sizing based on the outcome.

**Comment by KoenDG** [comment]
Changes implemented as requested.

File: test/framework/src/main/java/org/elasticsearch/test/InternalAggregationTestCase.java (hunk 4 above)
Comment (jasontedor) [inline test/framework/src/main/java/org/elasticsearch/test/InternalAggregationTestCase.java]: Note the extra reduce is added only if we are doing an extra reduce, and toReduce is greater than one.

Comment (KoenDG) [inline test/framework/src/main/java/org/elasticsearch/test/InternalAggregationTestCase.java]: Right, my bad, added it in.

**Comment by dakrone** [comment]
@koendg this needs to have master merged and the conflicts addressed and then ping Jason for another review

**Comment by jasontedor** [comment]
We are not going to pull this one in.

**Pull request closed without merging**